src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,554 @@
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    conditionals = [conditionals_b]
+    del device_b['installationParam']
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['measCapability']
+
+    # Create the actual config.
+    devices = [device_a, device_b]
+    config = {
+        'registrationRequests': devices,
+        'conditionalRegistrationData': conditionals,
+        'palRecords': pal_records,
+        'sasTestHarnessCert': os.path.join('certs', 'sas.cert'),
+        'sasTestHarnessKey': os.path.join('certs', 'sas.key')
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_2_default_config)
+  def test_WINNF_FT_S_PCR_2(self, config_filename):
+    """ Successful Maximum PPA Boundary Creation Clipped by Service Area Boundary.
+
+    Checks the maximum PPA boundary, being clipped by the Service Area composed of
+    one or more adjacent Census Tracts.
+    Checks PPA generated by SAS UUT shall be fully contained within the service area.
+    """
+
+    # Load the Config file.
+    config = loadConfig(config_filename)
+
+    # light checking of itu,terrain and landcover data path exists.
+    self.assertTrue(os.path.exists(DEFAULT_ITU_DATAPATH),
+                    msg='ITU Data path is not configured')
+    self.assertTrue(os.path.exists(DEFAULT_TERRAIN_DATAPATH),
+                    msg='Terrain Data path is not configured')
+    self.assertTrue(os.path.exists(DEFAULT_LANDCOVER_DATAPATH),
+                    msg='LandCover Data path is not configured')
+    self.assertTrue(os.path.exists(DEFAULT_CENSUSTRACTS_DATAPATH),
+                    msg='CensusTract Data path is not configured')
+
+    # Inject the PAL records.
+    for pal_record in config['palRecords']:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record)
+
+    # Register devices and  check response.
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # Trigger SAS UUT to create a PPA boundary.
+    pal_ids = [record['palId'] for record in config['palRecords']]
+    ppa_creation_request = {
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+    }
+
+    # Trigger PPA Creation to SAS UUT.
+    ppa_id = self.triggerPpaCreationAndWaitUntilComplete(ppa_creation_request)
+
+    # Notify SAS UUT about SAS Harness and trigger Full Activity Dump and retrieves the
+    # PPA Zone record.
+    uut_ppa_zone_data = self.triggerFadAndRetrievePpaZone(
+        ssl_cert=config['sasTestHarnessCert'],
+        ssl_key=config['sasTestHarnessKey'])
+
+    # Configure the Census Tract Directory that the PPA uses.
+    ppa.ConfigureCensusTractDriver(DEFAULT_CENSUSTRACTS_DATAPATH)
```

Comment:
Useless line (already setup as default), and btw function do not exist anymore. All drivers moved to geo/drive.py

Reply:
This   line is no longer required due to refactor the code based on   geo/drive.py     This line has been removed now.